lms/tasks/grading.py

```diff
@@ -0,0 +1,128 @@
+from sqlalchemy import select, exists
+
+from datetime import datetime
+from lms.models import (
+    GradingSyncGrade,
+    GradingSync,
+    LMSUser,
+    LTIRegistration,
+    Grouping,
+    ApplicationInstance,
+)
+from lms.services import LTIAHTTPService
+from lms.services.lti_grading.factory import LTI13GradingService
+from lms.tasks.celery import app
+
+
+@app.task()
+def sync_grades():
+    with app.request_context() as request:
+        with request.tm:
+            scheduled_syncs = request.db.scalars(
+                select(GradingSync)
+                .where(GradingSync.status == "scheduled")
+                .limit(5)
+                .with_for_update()
+            )
+
+            for sync in scheduled_syncs:
+                assignment = sync.assignment
+                assert (
+                    assignment.lis_outcome_service_url
+                ), "Assignment without grading URL"
+                lti_registration = request.db.scalars(
+                    select(LTIRegistration)
+                    .join(ApplicationInstance)
+                    .join(Grouping)
+                    .where(Grouping.id == assignment.course_id)
+                    .order_by(LTIRegistration.updated.desc())
+                ).first()
+                assert lti_registration, "No LTI registraion for LTI1.3 assignment"
+
+                for grade in sync.grades:
+                    sync_grade.delay(
+                        lti_registration_id=lti_registration.id,
+                        lis_outcome_service_url=assignment.lis_outcome_service_url,
+                        grading_sync_grade_id=grade.id,
+                        grading_sync_timestamp=sync.created.isoformat(),
+                    )
+
+                sync.status = "in_progress"
+
+
+@app.task()
+def sync_grades_complete(*, grading_sync_id):
```

Comment:
This look at the current state of individual grades to report back the status to the `GradingSync`, ie to mark it as completed.